oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/Analyzer.java

```diff
@@ -240,11 +242,23 @@ private void send(final AcceptableValue<?> v, final long time) {
     }
 
     private void generateTraffic(MeterEntity entity) {
-        ServiceTraffic s = new ServiceTraffic();
-        s.setName(requireNonNull(entity.getServiceName()));
-        s.setNodeType(NodeType.Normal);
-        s.setTimeBucket(TimeBucket.getMinuteTimeBucket(System.currentTimeMillis()));
-        MetricsStreamProcessor.getInstance().in(s);
+        if (entity.getDetectPoint() != null) {
+            switch (entity.getDetectPoint()) {
+                case SERVER:
+                    entity.setServiceName(entity.getDestServiceName());
+                    toService(requireNonNull(entity.getDestServiceName()));
+                    serverSide(entity);
+                    break;
+                case CLIENT:
+                    entity.setServiceName(entity.getSourceServiceName());
+                    toService(requireNonNull(entity.getSourceServiceName()));
+                    clientSide(entity);
+                    break;
+            }
```

Comment:
We should generate service traffic for both `source service` and `target service`? And that might be the reason for https://github.com/apache/skywalking/pull/7399/files#r683436684?

Reply:
envoy metric only has client-side DetectPoint, I think the others might only have server-side DetectPoint,
It's not about ` https://github.com/apache/skywalking/pull/7399/files#r683436684`.

Reply:
@kezhenxu94 This seems a logical decision only. Actually all traffic sources of services could be generated by the entity level metrics, rather than relation level.

Reply:
But consider in only 1% case, if someone only uses this, but disable another metrics set, the topology and service list could still show up. This is the only purpose.